panels/models/attraction.py

```diff
@@ -33,6 +34,30 @@ def sluggify(s):
     return RE_SLUG.sub('-', s).lower().strip('-')
 
 
+def treeify(items, keys, value_key=None):
```

Comment:
Didn't you do something similar to this for searching the schedule? Or am I misremembering, somehow...?

Reply:
For the schedule I made a `trie` which _is_ a tree, but a with a particular construction. This is _also_ making a tree, but with a different construction. Although I think a better name would be `groupify`, because that's the actual intent of the function. I'll make that change and add some docs!

😄